gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -404,6 +404,36 @@ void testTrinoQueryPropertiesTableExtraction(String query, Set<String> catalogs,
         assertThat(trinoQueryProperties.getCatalogs()).isEqualTo(catalogs);
     }
 
+    @Test
+    void testWithQueryNameExcluded()
+            throws IOException
+    {
+        String query = """
+                WITH dos (c1) as (select c1 from uno),
+                uno(c1) as (select c1 from cat.schem.tbl1)
+                SELECT c1 from uno, dos
```

Comment:
Uppercase as, select, from.

Reply:
i don't think this is valid SQL. You cannot use uno as it is defined after dos.

```suggestion
                WITH dos AS (SELECT c1 from cat.schem.tbl1),
                uno as (SELECT c1 FROM dos)
                SELECT c1 FROM uno, dos
```

you can test this in trino via simple sql 

```sql
// works fine
WITH dos AS (SELECT node_id FROM system.runtime.nodes),
uno AS (SELECT node_id as node_id2 FROM dos)
SELECT node_id, node_id2
FROM uno, dos

// doesn't work as uno is not defined before dos.
// Query 20241023_021715_04215_f8xaz failed: line 1:34: Table 'system.runtime.uno' does not exist
WITH dos AS (SELECT node_id FROM uno),
uno AS (SELECT node_id as node_id2 FROM system.runtime.nodes)
SELECT node_id, node_id2
FROM uno, dos
```